chemreg/compound/tests/test_models.py

```diff
@@ -45,6 +51,28 @@ def test_definedcompound_attr():
     assert inchikey.max_length == 29
 
 
+@pytest.mark.django_db(transaction=True)
+def test_query_structure_type():
+    """Test creation of `QueryStructureType` objects."""
+
+    # all the fields are non-nullable
+    with pytest.raises(IntegrityError):
+        QueryStructureTypeFactory(
+            name="ill-defined", label="Ill-Defined", short_description=None
+        )
+
+    # the name has to be free of reserved characters
+    qst1 = QueryStructureTypeFactory(name="ill defined", label="Ill Defined")
+    assert qst1.name == "ill-defined"
+
+    qst2 = QueryStructureTypeFactory(name="$omething weird")
+    assert qst2.name == "omething-weird"
+
+    # the name field needs to be unique
+    with pytest.raises(IntegrityError):
+        qst3 = QueryStructureTypeFactory(name="ill-defined", label="Ill Defined")
```

Comment:
```suggestion
    name = QueryStructureType._meta.get_field("name")
    assert name.unique
```
Example of non DB test.